File: test-npm-pip-pkg/py/tests/test_error_paths.py
import os
import sys
from utils.utils import client, run_test_section, log_result, base_url, organization_id
from imbrace import ImbraceClient, ApiError, AuthError

def test_error_paths():
    print("\n" + "=" * 70)
    print("🚀 STARTING: ERROR PATH TESTING (PY)")
    print("=" * 70)

    # 1. 404 Not Found
    def test_404():
        try:
            client.boards.get("non-existent-id-12345")
            raise Exception("Should have thrown 404")
        except ApiError as e:
            log_result("Caught Expected 404", str(e))
            if e.status_code != 404:
                print(f"   ⚠️ Expected 404 but got {e.status_code}")
    run_test_section("404 Not Found Handling", test_404)

    # 2. 401 Unauthorized
    def test_401():
        bad_client = ImbraceClient(
            api_key="invalid-key",
            base_url=base_url,
            organization_id="invalid-org"
        )
        try:
            bad_client.chat_ai.list_assistants()
            raise Exception("Should have thrown 401")
        except AuthError as e:
            log_result("Caught Expected 401", str(e))
    run_test_section("401 Unauthorized Handling", test_401)

    # 3. 400 Bad Request
    def test_400():
        try:
            # Missing required 'name' field
            client.boards.create(name="")
            # Note: some APIs might allow empty name, but usually they fail.
            # If this doesn't fail, we might need another 400 case.
            # Actually, calling create with nothing might not be possible with typed args in PY.
            # But we can pass empty dict if it was generic.
            # Let's try boards.create_field with invalid type
            client.boards.create_field("dummy", {"name": "Test", "type": "InvalidType"})
            raise Exception("Should have thrown 400")
        except ApiError as e:
            log_result("Caught Expected 400/404", str(e))
        except Exception as e:
            log_result("Caught Generic Error", str(e))
    run_test_section("400 Bad Request Handling", test_400)

    print("\n" + "=" * 70)
    print("✅ ERROR PATH TESTING COMPLETED")
    print("=" * 70)

if __name__ == "__main__":
    test_error_paths()

File: test-local-pkg/ts/tests/test-error-paths.ts
import { pathToFileURL } from 'url';
import { client, runTestSection, logResult } from "../utils/utils.js";

/**
 * TEST ERROR PATHS (SCENARIO-BASED)
 * Focuses on Error Handling: 401, 404, and network-like issues.
 */
async function testErrorPaths() {
  console.log("\n" + "=".repeat(70));
  console.log("🚀 STARTING: ERROR PATH TESTING");
  console.log("=".repeat(70));

  try {
    // --- Section 1: 404 Not Found ---
    await runTestSection("404 Not Found Handling", async () => {
      try {
        await client.boards.get("non-existent-id-12345");
        throw new Error("Should have thrown 404");
      } catch (e: any) {
        logResult("Caught Expected 404", e.message);
        if (e.statusCode !== 404 && !e.message.includes("404")) {
            console.warn("   ⚠️ Expected 404 but got different message:", e.message);
        }
      }
    });

    // --- Section 2: 401 Unauthorized (Mocking invalid client) ---
    await runTestSection("401 Unauthorized Handling", async () => {
      const { ImbraceClient } = await import("@imbrace/sdk");
      const badClient = new ImbraceClient({
          apiKey: "invalid-key",
          baseUrl: client.baseUrl,
          organizationId: "invalid-org"
      });

      try {
        await badClient.chatAi.listAiAgents();
        throw new Error("Should have thrown 401");
      } catch (e: any) {
        logResult("Caught Expected 401", e.message);
      }
    });

    // --- Section 3: Invalid Body/Payload (400) ---
    await runTestSection("400 Bad Request Handling", async () => {
      try {
        // Missing required 'name' field
        await client.boards.create({} as any);
        throw new Error("Should have thrown 400");
      } catch (e: any) {
        logResult("Caught Expected 400", e.message);
      }
    });

    console.log("\n" + "=".repeat(70));
    console.log("✅ ERROR PATH TESTING COMPLETED");
    console.log("=".repeat(70));

  } catch (err: any) {
    console.error("\n❌ FATAL TEST ERROR:", err.message);
    throw err;
  }
}

if (import.meta.url === pathToFileURL(process.argv[1]).href) {
  testErrorPaths().catch(console.error);
}

export { testErrorPaths };
